main.go

```diff
@@ -61,62 +53,169 @@ func List() {
 	Sort(entries)
 	Header()
 	for i, j := range entries {
-		fmt.Printf("%4d %14s %14s %20s\n", j.ID, j.LastName, j.FirstName, j.PhoneNumber)
+		fmt.Printf("%10d %20s %20s %20s\n", j.ID, j.LastName, j.FirstName, j.PhoneNumber)
 		if (i+1)%20 == 0 && i < len(entries) {
+			fmt.Println(" ")
 			fmt.Println("Press <ENTER> to continue...")
+			fmt.Println(" ")
 			fmt.Scanln()
 			Header()
 		}
 	}
+}
 
+func GiveID(entries []Entry) (i uint) {
+	i = uint(len(entries)) + 1 //Thats not ideal and will work wrong if the last contact will be deleted
+	return i
 }
 
-func Add(entries []Entry) {
-	// how to encode with user input???
-	f, err := os.Open("phonebook.json")
+func Add() {
+	// how to encode with user input??? Tutorial by me (version 1)
+	var entries []Entry // make variable where will be all data
+
+	f, err := os.Open("phonebook.json") // open and encode file with already been information and put it in variable you made in previous step
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	if err := json.NewDecoder(bufio.NewReader(f)).Decode(&entries); err != nil {
+		log.Fatal(err)
+	}
+
+	var newentries Entry // make new variable with type of your struct, but be careful - it need to be NOT SLICE OF STRUCT, JUST STRUCT, CAUSE WE MAKING ONLY ONE NEW CONTACT, NOT 5 IN ONE TIME
+
+	scanner := bufio.NewScanner(os.Stdin) // Scan information you need
```

Comment:
If you want to use `bufio.Scanner`, you have to create it once in `main`, and pass to all these functions, where you may need it. It's actually better that way (when you pass it)